lib/modules/manager/vendir/artifacts.spec.ts

```diff
@@ -0,0 +1,509 @@
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache',
+    );
+    fs.getParentDir.mockReturnValue('');
+
+    // artifacts
+    fs.getSiblingFileName.mockReturnValueOnce('vendor');
+    git.getRepoStatus.mockResolvedValueOnce(
+      partial<StatusResult>({
+        not_added: ['vendor/Chart.yaml'],
+      }),
+    );
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await vendir.updateArtifacts({
+        packageFileName: 'vendir.yml',
+        updatedDeps,
+        newPackageFileContent: vendirFile,
+        config: {
+          ...config,
+        },
+      }),
+    ).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'vendir.lock.yml',
+          contents: vendirLockFile2,
+        },
+      },
+      {
+        file: {
+          type: 'addition',
+          path: 'vendor/Chart.yaml',
+          contents: undefined,
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
+      {
+        cmd: 'vendir sync',
+        options: {
+          env: {
+            HOME: '/home/user',
+            HTTPS_PROXY: 'https://example.com',
+            HTTP_PROXY: 'http://example.com',
+            LANG: 'en_US.UTF-8',
+            LC_ALL: 'en_US',
+            NO_PROXY: 'localhost',
+            PATH: '/tmp/path',
+          },
+        },
+      },
+    ]);
+  });
+
+  it('works explicit global binarySource', async () => {
+    GlobalConfig.set({ ...adminConfig, binarySource: 'global' });
+    fs.readLocalFile.mockResolvedValueOnce(vendirLockFile1);
+    fs.getSiblingFileName.mockReturnValueOnce('vendir.lock.yml');
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce(vendirLockFile2);
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache',
+    );
+    fs.getParentDir.mockReturnValue('');
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await vendir.updateArtifacts({
+        packageFileName: 'vendir.yml',
+        updatedDeps,
+        newPackageFileContent: vendirFile,
+        config,
+      }),
+    ).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'vendir.lock.yml',
+          contents: vendirLockFile2,
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([{ cmd: 'vendir sync' }]);
+  });
+
+  it('supports install mode', async () => {
+    GlobalConfig.set({
+      ...adminConfig,
+      binarySource: 'install',
+    });
+    fs.readLocalFile.mockResolvedValueOnce(vendirLockFile1);
+    fs.getSiblingFileName.mockReturnValueOnce('vendir.lock.yml');
+    fs.readLocalFile.mockResolvedValueOnce(vendirLockFile2);
+    fs.readLocalFile.mockResolvedValueOnce('0.35.0');
+    const execSnapshots = mockExecAll();
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache',
+    );
+    datasource.getPkgReleases.mockResolvedValueOnce({
```

Comment:
you can remove this when you're using constraints below, renovate doesn't call it if constraints are single versions.